safety/scan/main.py

```diff
@@ -211,13 +253,64 @@ def process_files(paths: Dict[str, Set[Path]], config: Optional[ConfigModel] = N
     if not config:
         config = ConfigModel()
 
-    for file_type_key, f_paths in paths.items():
-        file_type = FileType(file_type_key)
-        if not file_type or not file_type.ecosystem:
-            continue
-        for f_path in f_paths:
-            with InspectableFileContext(f_path, file_type=file_type) as inspectable_file:
-                if inspectable_file and inspectable_file.file_type:
-                    inspectable_file.inspect(config=config)
-                    inspectable_file.remediate()
-                    yield f_path, inspectable_file
+    # old GET implementation
+    if not use_server_matching:
+        for file_type_key, f_paths in paths.items():
+            file_type = FileType(file_type_key)
+            if not file_type or not file_type.ecosystem:
+                continue
+            for f_path in f_paths:
+                with InspectableFileContext(f_path, file_type=file_type) as inspectable_file:
+                    if inspectable_file and inspectable_file.file_type:
+                        inspectable_file.inspect(config=config)
+                        inspectable_file.remediate()
+                        yield f_path, inspectable_file
+
+    # new POST implementation
+    else:
+        files_metadata = []
+        repo_path = os.getcwd()
+        meta = build_meta(repo_path)
+        for file_type_key, f_paths in paths.items():
+            file_type = FileType(file_type_key)
+            if not file_type or not file_type.ecosystem:
+                continue
+            for f_path in f_paths:
+                relative_path = os.path.relpath(f_path, start=os.getcwd())
+                # Read the file content
+                try:
+                    with open(f_path, "r") as file:
+                        content = file.read()
+                except Exception as e:
+                    LOG.error(f"Error reading file {f_path}: {e}")
+                    continue
+                # Append metadata to the payload
+                files_metadata.append({
+                    "name": relative_path,
+                    "content": content,
+                })
+
+        # Prepare the payload with metadata at the top level
+        payload = {
+            "meta": meta,
+            "files_metadata": files_metadata,
+        }
+
+        response = obj.auth.client.upload_requirments(payload)
+
+        if response.status_code == 201:
+            LOG.info("Scan Payload successfully sent to the API.")
+        else:
+            LOG.error(f"Failed to send scan payload to the API. Status code: {response.status_code}")
+            LOG.error(f"Response: {response.text}")
+
+        for file_type_key, f_paths in paths.items():
```

Comment:
This code is probably not needed for this execution.

One thing that we should be able to do is instead of returning the `InspectableFileContext` is to introduce new CLI only model that will expose the same attributes as the `InspectableFileContext` but we will be able to return it populated from API response.

Reply:
We can remove it, but for this V1 it is focussed on showing the client-server connection but does not use that data yet so this may cause it not to run fully when running `safety scan --use-server-matching`

Reply:
removed